File: vben-admin/apps/web-antd/src/api/class.ts
import { requestClient } from '#/api/request';

export interface ClassItem {
  cid: number;
  name: string;
  price: string;
  content?: string;
  fenlei?: string;
  docking?: string;
  noun?: string;
  status: number;
}

export interface ClassCategory {
  id: number;
  name: string;
  recommend?: number;
  log?: number;
  ticket?: number;
  changepass?: number;
  allowpause?: number;
}

export interface ClassListPagedResult {
  list: ClassItem[];
  pagination: {
    has_more: boolean;
    limit: number;
    page: number;
    total: number;
  };
}

export interface CourseItem {
  id: string;
  name: string;
  kcjs?: string;
  studyStartTime?: string;
  studyEndTime?: string;
  examStartTime?: string;
  examEndTime?: string;
  learnStatusName?: string;
  complete?: string;
  idx?: number;
  select?: boolean;
}

export interface CourseQueryResult {
  data: CourseItem[];
  msg: string;
  userinfo: string;
  userName: string;
}

/** 获取课程列表 */
export async function getClassListApi(params?: {
  fenlei?: number;
  search?: string;
}) {
  return requestClient.get<ClassItem[]>('/class/list', { params });
}

/** 获取分页课程列表 */
export async function getClassListPagedApi(params?: {
  favorite?: number;
  fenlei?: number;
  limit?: number;
  page?: number;
  search?: string;
}) {
  return requestClient.get<ClassListPagedResult>('/class/list-paged', {
    params,
  });
}

/** 获取课程分类 */
export async function getClassCategoriesApi() {
  return requestClient.get<ClassCategory[]>('/class/categories');
}

/** 查课 */
export async function queryCourseApi(cid: number, userinfo: string) {
  return requestClient.post<CourseQueryResult>('/class/search', {
    cid,
    userinfo,
  });
}

/** 获取课程所属分类的开关配置 */
export async function getCategorySwitchesApi(cid: number) {
  return requestClient.get<{
    allowpause: number;
    changepass: number;
    log: number;
    ticket: number;
  }>('/class/category-switches', { params: { cid } });
}

/** 下单 */
export async function addOrderApi(data: {
  cid: number;
  data: Array<{ data: CourseItem; userinfo: string; userName: string }>;
  remarks?: string;
}) {
  return requestClient.post('/order/add', data);
}

File: go-api/internal/model/class.go
package model

type Class struct {
	CID     int    `json:"cid" db:"cid"`
	Name    string `json:"name" db:"name"`
	Noun    string `json:"noun" db:"noun"`
	Price   string `json:"price" db:"price"`     // DB: varchar
	Docking string `json:"docking" db:"docking"` // DB: varchar
	Fenlei  string `json:"fenlei" db:"fenlei"`   // DB: varchar
	Status  int    `json:"status" db:"status"`
	Sort    int    `json:"sort" db:"sort"`
	Content string `json:"content,omitempty"`
}

type Category struct {
	ID                int    `json:"id" db:"id"`
	Name              string `json:"name" db:"name"`
	Sort              int    `json:"sort" db:"sort"`
	Status            string `json:"status" db:"status"`
	Time              string `json:"time" db:"time"`
	Recommend         int    `json:"recommend" db:"recommend"`
	Log               int    `json:"log" db:"log"`
	Ticket            int    `json:"ticket" db:"ticket"`
	ChangePass        int    `json:"changepass" db:"changepass"`
	AllowPause        int    `json:"allowpause" db:"allowpause"`
	SupplierReport    int    `json:"supplier_report" db:"supplier_report"`
	SupplierReportHID int    `json:"supplier_report_hid" db:"supplier_report_hid"`
}

type CategoryListRequest struct {
	Page    int    `form:"page"`
	Limit   int    `form:"limit"`
	Keyword string `form:"keyword"`
	Status  string `form:"status"`
}

type ClassListRequest struct {
	Fenlei   int    `form:"fenlei"`
	Status   int    `form:"status"`
	Search   string `form:"search"`
	Page     int    `form:"page"`
	Limit    int    `form:"limit"`
	Favorite int    `form:"favorite"`
}

// ===== 查课/下单相关 =====

type CourseQueryRequest struct {
	CID      int    `json:"cid" binding:"required"`
	UserInfo string `json:"userinfo" binding:"required"`
}

type CourseItem struct {
	ID             string `json:"id"`
	Name           string `json:"name"`
	KCJS           string `json:"kcjs,omitempty"`
	StudyStartTime string `json:"studyStartTime,omitempty"`
	StudyEndTime   string `json:"studyEndTime,omitempty"`
	ExamStartTime  string `json:"examStartTime,omitempty"`
	ExamEndTime    string `json:"examEndTime,omitempty"`
	Complete       string `json:"complete,omitempty"`
}

type CourseQueryResponse struct {
	UserInfo string       `json:"userinfo"`
	UserName string       `json:"userName"`
	Msg      string       `json:"msg"`
	Data     []CourseItem `json:"data"`
}

// SupplierQueryResult 上游供应商查课响应
type SupplierQueryResult struct {
	Code     int          `json:"code"`
	Msg      string       `json:"msg"`
	UserName string       `json:"userName"`
	Data     []CourseItem `json:"data"`
}

// SupplierOrderResult 上游供应商下单响应
type SupplierOrderResult struct {
	Code int    `json:"code"`
	Msg  string `json:"msg"`
	YID  string `json:"id,omitempty"`
}

// SupplierProgressItem 上游供应商进度查询响应项 (按 PHP jdjk.php processCx 返回的 data 数组元素)
type SupplierProgressItem struct {
	YID             string `json:"id"`
	Noun            string `json:"noun"`
	KCName          string `json:"kcname"`
	User            string `json:"user"`
	Status          string `json:"status"`
	StatusText      string `json:"status_text"`
	Process         string `json:"process"`
	Remarks         string `json:"remarks"`
	CourseStartTime string `json:"courseStartTime"`
	CourseEndTime   string `json:"courseEndTime"`
	ExamStartTime   string `json:"examStartTime"`
	ExamEndTime     string `json:"examEndTime"`
}

// SupplierBatchProgressRef 为供应商级批量进度拉取提供最小订单上下文。
type SupplierBatchProgressRef struct {
	YID    string `json:"yid"`
	User   string `json:"user"`
	KCName string `json:"kcname"`
	KCID   string `json:"kcid"`
	Noun   string `json:"noun"`
}
